coreos-installer

```diff
@@ -513,6 +513,137 @@ get_ignition_url() {
     done
 }
 
+############################################################
+#Persist network configuration
+############################################################
+urlencode() {
+    local string="$( cat ${1})"
+    local strlen=${#string}
+    local encoded=""
+    local pos c o
+
+    for (( pos=0 ; pos<strlen ; pos++ )); do
+        c=${string:$pos:1}
+        case "$c" in
+            [-_.~a-zA-Z0-9] ) o="${c}" ;;
+            * )               printf -v o '%%%02x' "'$c"
+        esac
+        encoded+="${o}"
+    done
+    url_encoded_file="${encoded}"
+}
+
+write_ifcfg_files() {
+    local IPS NAMESERVERS
+    if [ ! -f /tmp/ip ]; then
+        log "Not persisting network configs; none provided"
+        return
+    fi
+    type ip_to_var >/dev/null 2>&1 || . /lib/net-lib.sh
+    type strglobin >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+    IPS=($(cat /tmp/ip))
+    # Skip if no ips specified
+    [ ${#IPS[@]} == "0" ] && continue
+
+    log "Persisting provided network configs"
+    mkdir -p /tmp/ifcfg/
+
+    # Prepare nameservers list
+    NAMESERVERS=()
+    if [ -f /tmp/nameserver ]; then
+        NAMESERVERS=($(cat /tmp/nameservers))
+    fi
+
+    echo '{"ignition": {"version": "2.2.0"}, "storage": { "files": [' > /tmp/network.ign
```

Comment:
Hardcoding 2.2.0 version is probably a bad idea, although FCOS should be able to parse that, right?

Reply:
No it can't.